### PR TITLE
Update connections on BindingRepository.Write

### PR DESCRIPTION
**Pull request by georgii-borovinskikh-sonarsource**

### CODE REVIEW

File: src/ConnectedMode.UnitTests/Persistence/SolutionBindingRepositoryTests.cs [hunk 1]
```diff
@@ -170,6 +170,18 @@ public void Write_FileNotWritten_CredentialsNotWritten()
 
             credentialsLoader.Verify(x => x.Save(It.IsAny<ICredentials>(), It.IsAny<Uri>()), Times.Never);
         }
+        
+        [TestMethod]
+        public void Write_FileNotWritten_EventNotTriggered()
```
Comment (ugras-ergun-sonarsource) [inline src/ConnectedMode.UnitTests/Persistence/SolutionBindingRepositoryTests.cs:175]: The new two tests can be merged into one with parameters

**Review by ugras-ergun-sonarsource — APPROVED** [review]

File: src/Core/Binding/ISolutionBindingRepository.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    /// <summary>
+    /// Writes the binding information
+    /// </summary>
+    /// <returns>Has file been saved</returns>
+    bool Write(string configFilePath, BoundSonarQubeProject binding);
+
+    /// <summary>
+    /// 
+    /// </summary>
```
Comment (pavel-mikula-sonarsource) [inline src/Core/Binding/ISolutionBindingRepository.cs:42]: This either shouldn't be here, or should have content
```suggestion
```

File: src/SLCore/State/AliveConnectionTracker.cs [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+            var connectionId = connectionIdHelper.GetConnectionIdFromUri(serverUri, organization);
+
+            if (serverUri == ConnectionIdHelper.SonarCloudUri)
+            {
+                connections[connectionId] = new SonarCloudConnectionConfigurationDto(connectionId, true, organization);
+            }
+            else
+            {
+                connections[connectionId] = new SonarQubeConnectionConfigurationDto(connectionId, true, serverUri.ToString());
+            }
```
Comment (pavel-mikula-sonarsource) [inline src/SLCore/State/AliveConnectionTracker.cs:124]: This should be an assignment from ternary, to structurally ensure that the assigned thing is always the same
```
connections[connectionId] =serverUri == ConnectionIdHelper.SonarCloudUri
    ? new SonarCloudConn...
    : new SonarQubeConn...

```

File: src/SLCore/State/AliveConnectionTracker.cs [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+using SonarLint.VisualStudio.SLCore.Common.Helpers;
+using SonarLint.VisualStudio.SLCore.Core;
+using SonarLint.VisualStudio.SLCore.Service.Connection;
+using SonarLint.VisualStudio.SLCore.Service.Connection.Models;
+
+namespace SonarLint.VisualStudio.SLCore.State;
+
+/// <summary>
+/// 
+/// </summary>
```
Comment (pavel-mikula-sonarsource) [inline src/SLCore/State/AliveConnectionTracker.cs:39]: Noise
```suggestion
```

**Pull request merged**